src/common/components/repository-input/index.js

```diff
@@ -0,0 +1,70 @@
+import 'isomorphic-fetch';
+
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import {
+  changeRepositoryInput,
+  verifyGitHubRepository
+} from '../../actions/repository-input';
+
+export class RepositoryInput extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  getStatusMessage() {
+    const input = this.props.repositoryInput;
+    let message;
+
+    if (!input.repository) {
+      message = 'Invalid repository URL.';
+    } else if (input.repository && input.isFetching) {
+      message = `Verifying ${input.repository} on GitHub...`;
+    } else if (input.success && input.repositoryUrl) {
+      message = `Repository ${input.repository} is valid.`;
+    } else if (input.errors) {
+      message = `Repository ${input.repository} is invalid.`;
```

Comment:
I think we need to distinguish here between an invalid GitHub repository name ("081 811 8181" / "katie.fenn@googlemail.com") and an inaccessible / non-existant GitHub repository ("katiefenn/thisrepodoesntexist").

Suggested message: "Repository either does not exist, or access not authorised."